hw/ip_templates/rv_plic/fpv/tb/coverage_waivers.tcl.tpl

```diff
@@ -61,3 +61,27 @@ assert -name FSMParasiticState::AlertSenderFSMParasiticState_A\
  {!(dut.gen_alert_tx[0].u_prim_alert_sender.state_q inside\
   {Idle, AlertHsPhase1, AlertHsPhase2, PingHsPhase1, PingHsPhase2, Pause0, Pause1}) ->\
   dut.gen_alert_tx[0].u_prim_alert_sender.state_d == Idle}
+
+proc clog2 {NumSrc} {
+  return [expr {ceil(log($NumSrc)/log(2))}]
+}
+
+# These are all the dead nodes in a binary tree. They are dead because the rightmost nodes at the
+# bottom of the tree are tied with 1'b0. Hence, their parents on levels below them also assigned
+# with 1'b0.
+proc tree {NumSrc} {
+  set Numlevels [clog2 $NumSrc]
+  for {set level 1} {$level <= $Numlevels-1} {incr level} {
```

Comment:
Would it be easier to use `$level < $Numlevels`? (It looks a bit more like C that way! :-D)